sympy/functions/elementary/exponential.py

```diff
@@ -541,6 +541,8 @@ def _eval_as_leading_term(self, x, logx=None, cdir=0):
             # Check out function: test_issue_18473() in test_exponential.py and
             # test_limits.py for more information.
             return exp(arg0)
+        if isinstance(arg.as_numer_denom()[0], log):
```

Comment:
It seems that this will only work in very special cases not including this:
```
>>> O(x**(2*x), (x, oo))
O(exp(x*log(1/x)), (x, oo))
```

Reply:
Yupp I am fixing this , i realized this will become a bit specialized . Hence I turned this into draft . Working on this